PhysicsTools/Heppy/python/analyzers/objects/JetAnalyzer.py

```diff
@@ -83,13 +109,23 @@ def __init__(self, cfg_ana, cfg_comp, looperName):
           if   doResidual == "MC":   doResidual = self.cfg_comp.isMC
           elif doResidual == "Data": doResidual = not self.cfg_comp.isMC
           elif doResidual not in [True,False]: raise RuntimeError, "If specified, applyL2L3Residual must be any of { True, False, 'MC', 'Data'(default)}"
-          GT = getattr(cfg_comp, 'jecGT', mcGT if self.cfg_comp.isMC else dataGT)
+          #GT = getattr(cfg_comp, 'jecGT', mcGT if self.cfg_comp.isMC else dataGT)
+          GTs = []
+          if self.cfg_comp.isMC:
+              GTs.append( getattr(cfg_comp, 'jecGT', mcGT) )
+              #print str(mcGT), getattr(cfg_comp, 'jecGT', mcGT)
+          else:
+              for gt in dataGT:
+                  GTs.append( getattr(cfg_comp, 'jecGT', gt) )
```

Comment:
I think a more proper format for run-dependent JECs should be either a list of IOVs `((first run, last run,), GT)` or maybe better just `(first run, GT)` so that the last IOV is always open-ended.
Having two separate lists, one of GTs and one of runs, and relying on them being aligned by index, seems really error prone to me.